sdk/core/azure-core/src/main/java/com/azure/android/core/util/function/Function.java

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util.function;
+
+/**
+ * Simple interface make code compatible with Java 7
+ */
+public interface Function<Input, Output> {
```

Comment:
We should not duplicate the Java 8 interface types if possible. I know that Android doesn't contain the function types, but since the Android toolchain can compile code with Lambdas, there has to be some Type or Interface it uses to represent them. Can we see what type that is?

Reply:
Did some research and it seems that the AndroidX has a [package](https://developer.android.com/reference/androidx/arch/core/util/package-summary) for this.

Reply:
According to our discussion today, we will have a deeper discussion about the how we access platform data, which will involve the Configuration class (the only one using this interface so far). Depending on how this goes we might not even need to rely on AndroidX.